apps/nn_ops/Makefile

```diff
@@ -2,7 +2,55 @@ include ../support/Makefile.inc
 
 BIN ?= bin
 
-all: $(BIN)/host/MatrixMultiply
+all: $(BIN)/host/AveragePool $(BIN)/host/Convolution $(BIN)/host/DepthwiseConvolution $(BIN)/host/Im2col $(BIN)/host/MatrixMultiply $(BIN)/host/MaxPool
+
+$(BIN)/AveragePool.generator: AveragePool_generator.cpp common.cpp $(GENERATOR_DEPS)
+	@mkdir -p $(@D)
+	$(CXX) $(CXXFLAGS) -g -fno-rtti $(filter-out %.h,$^) -o $@ $(LDFLAGS) $(HALIDE_SYSTEM_LDFLAGS)
+
+$(BIN)/%/AveragePool.o: $(BIN)/AveragePool.generator
+	@mkdir -p $(@D)
+	$^ -g AveragePool -o $(BIN)/$* -e o,h -f AveragePool target=$(HL_TARGET)
+
+$(BIN)/%/AveragePool: AveragePool.cpp $(BIN)/%/AveragePool.o
+	@mkdir -p $(@D)
+	$(CXX-$*) $(CXXFLAGS) $(CXXFLAGS-$*) -I $(BIN)/$* -Wall -O3 AveragePool.cpp $(BIN)/$*/AveragePool.o -o $(BIN)/$*/AveragePool $(LDFLAGS-$*)
+
+$(BIN)/Convolution.generator: Convolution_generator.cpp common.cpp $(GENERATOR_DEPS)
+	@mkdir -p $(@D)
+	$(CXX) $(CXXFLAGS) -g -fno-rtti $(filter-out %.h,$^) -o $@ $(LDFLAGS) $(HALIDE_SYSTEM_LDFLAGS)
+
+$(BIN)/%/Convolution.o: $(BIN)/Convolution.generator
+	@mkdir -p $(@D)
+	$^ -g Convolution -o $(BIN)/$* -e o,h -f Convolution target=$(HL_TARGET)
+
+$(BIN)/%/Convolution: Convolution.cpp $(BIN)/%/Convolution.o
+	@mkdir -p $(@D)
+	$(CXX-$*) $(CXXFLAGS) $(CXXFLAGS-$*) -I $(BIN)/$* -Wall -O3 Convolution.cpp $(BIN)/$*/Convolution.o -o $(BIN)/$*/Convolution $(LDFLAGS-$*)
+
+$(BIN)/DepthwiseConvolution.generator: DepthwiseConvolution_generator.cpp common.cpp $(GENERATOR_DEPS)
+	@mkdir -p $(@D)
+	$(CXX) $(CXXFLAGS) -g -fno-rtti $(filter-out %.h,$^) -o $@ $(LDFLAGS) $(HALIDE_SYSTEM_LDFLAGS)
+
+$(BIN)/%/DepthwiseConvolution.o: $(BIN)/DepthwiseConvolution.generator
+	@mkdir -p $(@D)
+	$^ -g DepthwiseConvolution -o $(BIN)/$* -e o,h -f DepthwiseConvolution target=$(HL_TARGET)
+
+$(BIN)/%/DepthwiseConvolution: DepthwiseConvolution.cpp $(BIN)/%/DepthwiseConvolution.o
```

Comment:
Looks like we're only building/testing one depth_multiplier (1) ?

Reply:
Good point, added some more.